flocker/node/test/test_deploy.py

```diff
@@ -3044,6 +3095,52 @@ def test_return(self):
             _to_volume_name(volume.dataset.dataset_id)))
 
 
+class DeleteDatasetTests(TestCase):
+    """
+    Tests for ``DeleteDataset``.
+    """
+    def setUp(self):
+        self.volume_service = create_volume_service(self)
+        self.deployer = P2PNodeDeployer(
+            u'example.com',
+            self.volume_service,
+            docker_client=FakeDockerClient(),
+            network=make_memory_network())
+
+        id1 = unicode(uuid4())
+        self.volume1 = self.volume_service.get(_to_volume_name(id1))
+        id2 = unicode(uuid4())
+        self.volume2 = self.volume_service.get(_to_volume_name(id2))
+        self.successResultOf(self.volume_service.create(self.volume1))
+        self.successResultOf(self.volume_service.create(self.volume2))
+
+    def test_creates(self):
+        """
+        ``CreateDataset.run()`` deletes volumes whose ``dataset_id`` matches
+        the one the instance was created with.
+        """
+        delete = DeleteDataset(
+            dataset=Dataset(dataset_id=self.volume2.name.dataset_id))
+        self.successResultOf(delete.run(self.deployer))
+
+        self.assertEqual(
+            list(self.successResultOf(self.volume_service.enumerate())),
+            [self.volume1])
+
+    @validate_logging(
+        lambda test, logger: logger.flush_tracebacks(RuntimeError))
+    def test_failed_create(self, logger):
+        """
+        Failed deletions of volumes are swallowed.
+        """
+        self.patch(self.volume_service.pool, "destroy",
+                   lambda fs: fail(RuntimeError()))
```

Comment:
nit, `RuntimeError` is for errors in the runtime (ie, CPython), not errors that happen at run time.